contracts/core/extensions/CoreInternal.sol

```diff
@@ -149,18 +160,27 @@ contract CoreInternal is
      * Can only be called by owner of Core.
      *
      * @param _exchangeId   Enumeration of exchange within the mapping
+     * @param _exchange     Address of the exchange conforming to IExchangeWrapper
      */
     function removeExchange(
-        uint8 _exchangeId
+        uint8 _exchangeId,
+        address _exchange
     )
         external
         onlyOwner
     {
-        state.exchanges[_exchangeId] = address(0);
+        require(
+            state.exchangeIds[_exchangeId] != address(0),
+            "CoreInternal.removeExchange: Exchange already removed"
+        );
+
```

Comment:
Do we need to check that the _exchangeID matches the exchange address? Otherwise can't we put in an arbitrary exchange?